Scene manager: Show statuses

iLLiCiTiT opened the pull request:
## Changelog Description
Added statuses to scene inventory manager.

## Additional info
The code of scene inventory tool was not much clear so it was "easier" to refactor some parts for easier understanding where and how data are comming in. The refactor is done only in bounds of this feature. Also discovered few bugs that probably caused issues even in current develop, if you hit a bug please make sure it is bug related to this PR.

## Screenshot
![image](https://github.com/ynput/ayon-core/assets/43494761/7f26c067-075c-417b-95fb-c10edc2d3f25)
![image](https://github.com/ynput/ayon-core/assets/43494761/1001acc4-7eca-4c4d-b5d2-5ef26ef9fc2f)


## Testing notes:
1. Launch host of your choice.
2. Load/reference some representations in.
3. Scene inventory should show them.
4. You should be able to change their version, update and remove them, switch to hero and back, switch to different representation.
5. Status should show current status of version of loaded representation.

BigRoy commented on client/ayon_core/tools/sceneinventory/select_version_dialog.py:
```diff
@@ -0,0 +1,218 @@
+    def _on_confirm(self):
+        self._cancelled = False
+        index = self._versions_combobox.currentIndex()
+        print(index)
+        item_id = self._versions_combobox.itemData(index, ITEM_ID_ROLE)
+        print(item_id)
```
```suggestion
        item_id = self._versions_combobox.itemData(index, ITEM_ID_ROLE)
```

LiborBatek reviewed (changes requested):
I have started testing it in ```Maya``` host app and have some first issues:

1) ```Update All``` isnt working atm

<img width="526" alt="Screenshot 2024-05-30 085014" src="https://github.com/ynput/ayon-core/assets/112623825/0d22d2bf-c171-445a-9e46-c5d83772472d">


2) ```Filter outdated``` isnt working atm

3) ```Hero version``` doesnt propagate status from the origin for ie ```v003``` and ```[v003]``` doesnt match status


 
![Screenshot 2024-05-30 085113](https://github.com/ynput/ayon-core/assets/112623825/2c1f38ec-fc83-4667-834e-02fc50234635)

iLLiCiTiT commented:
> 1. Update All isnt working atm

Fixed

> 2. Filter outdated isnt working atm

Fixed

> 3. Hero version doesnt propagate status from the origin for ie v003 and [v003] doesnt match status

Hero version has own status, it is not related to origin version.

LiborBatek reviewed (approved):
Now even ```Filter outdated``` and ```Update all outdated``` works ok.

I have also did some additional testing within ```Blender``` host app and all good too!

LiborBatek commented:
If something to note, there is a slight UX issue with Qt formating or so...

Could be aligned to ```Version``` column imho all the time
![ScnInv](https://github.com/ynput/ayon-core/assets/112623825/db083ec2-249a-4d62-a5d6-01052ce8c980)
